client/allocrunner/taskrunner/script_check_hook.go

```diff
@@ -119,72 +79,67 @@ func (h *scriptCheckHook) Name() string {
 	return "script_checks"
 }
 
-// PostStart implements interfaces.TaskPoststartHook. It adds the current
-// task context (driver and env) to the script checks and starts up the
-// scripts.
+// Prestart implements interfaces.TaskPrestartHook. It stores the
+// initial structs.Task
+func (h *scriptCheckHook) Prestart(ctx context.Context, req *interfaces.TaskPrestartRequest, _ *interfaces.TaskPrestartResponse) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.task = req.Task
+	return nil
+}
+
+// PostStart implements interfaces.TaskPoststartHook. It creates new
+// script checks with the current task context (driver and env), and
+// starts up the scripts.
 func (h *scriptCheckHook) Poststart(ctx context.Context, req *interfaces.TaskPoststartRequest, _ *interfaces.TaskPoststartResponse) error {
 	h.mu.Lock()
 	defer h.mu.Unlock()
 
 	if req.DriverExec == nil {
 		return fmt.Errorf("driver doesn't support script checks")
```

Comment:
Sorry this is unrelated but won't this fail any task that uses a driver that doesn't support script checks?

Reply:
Yes, looks like it. This should be pushed under the `newScriptCheck` and return `nil`.

Reply:
I'll see about adding a dummy `tcp` check to the e2e testing... we had no e2e coverage of checks. 😦

Reply:
Fixed in da342e2